src/bundle.c

```diff
@@ -723,6 +727,53 @@ static int install_bundles(struct list *bundles, struct list **subs, int current
 	to_install_files = filter_out_existing_files(to_install_files);
 	grabtime_stop(&times);
 
+	/* Step 1.5: Now that we know the bundle is valid, check if we have enough space */
+	if (!skip_diskspace_check) {
+		char *filepath = NULL;
+		ptr = list_head(to_install_bundles);
+		while (ptr) {
+			struct manifest *subman;
+			subman = ptr->data;
+			ptr = ptr->next;
+
+			struct bundle_result *bundle = NULL;
+			bundle = calloc(sizeof(struct bundle_result), 1);
+			/* point to or copy relevant information over for future use */
+			strncpy(bundle->bundle_name, subman->component, BUNDLE_NAME_MAXLEN - 1);
+			bundle->topsize = subman->contentsize;
+			bundle->includes = subman->includes;
+			bundle_info = list_prepend_data(bundle_info, bundle);
+			bundle_size += calculate_size(subman->component, bundle_info, false);
+		}
+		list_free_list_and_data(bundle_info, free);
+
+		/* default is '/', so if '/', check 'usr', else, use the custom path. */
+		if(strcmp("/", path_prefix) == 0) {
```

Comment:
You can just use  `string_or_die(&filepath, "%s/%s", path_prefix, "/usr/");` for the default path too. It's not a problem to have "//usr" as a path and the code gets easier to read.

Reply:
Done, but using mk_full_filename() as suggested above.